engineioxide/src/engine.rs

```diff
@@ -83,7 +85,15 @@ impl<H: EngineIoHandler> EngineIo<H>
         self.handler.on_connect(&socket);
 
         let packet = OpenPacket::new(TransportType::Polling, sid, &self.config);
-        let packet: String = Packet::Open(packet).try_into()?;
+        #[allow(unused_mut)]
+        let mut packet: String = Packet::Open(packet).try_into()?;
+        cfg_if! {
+            if #[cfg(feature = "v3")] {
+                if protocol == ProtocolVersion::V3 {
```

Comment:
It would be nice to add some comments about difference between v4 and V3 when you are doing custom protocol version actions